### PR TITLE
IBX-5221: Updated `Attribute` REST structure

### PR DESCRIPTION
**Pull request by konradoboza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-5221](https://issues.ibexa.co/browse/IBX-5221)
| Versions      | `4.4`, `main`
| Edition       | Commerce

REST reference update originating from https://github.com/ibexa/product-catalog/pull/915.

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

**Review by Steveb-p — APPROVED** [review]

File: docs/api/rest_api/rest_api_reference/input/examples/product/catalog/attributes/GET/AttributeList.json.example [hunk 1]
```diff
@@ -8,29 +8,32 @@
                 "name": "Attrib1",
                 "description": "",
                 "type": "Checkbox",
+                "type_identifier": "checkbox",
                 "group": "Attribute Group Test",
                 "position": 0,
-                "options": ""
+                "options": []
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/examples/product/catalog/attributes/GET/AttributeList.json.example:14]: A bit sideway question: Why this isn't `"options": {} ` as it seems to be an object when not empty? Empty array VS associative array encoding to JSON maybe?

Comment (konradoboza) [reply docs/api/rest_api/rest_api_reference/input/examples/product/catalog/attributes/GET/AttributeList.json.example:14]: Yes, most likely. This comes from `Ibexa\Contracts\Rest\Output\Generator::valueElement` on `array<string,mixed>`.

Comment (adriendupuis) [reply docs/api/rest_api/rest_api_reference/input/examples/product/catalog/attributes/GET/AttributeList.json.example:14]: @konradoboza 

With all those cascades and recursion, it's not as easy as to add a `JSON_FORCE_OBJECT`… 

Anyway, it was a side remark.

**Review by adriendupuis — APPROVED** [review]

**Comment by konradoboza** [comment]
I pushed a small update reflecting the recent changes in the corresponding PR, ref: https://github.com/ibexa/product-catalog/commit/a7e0235b56b9946042f2c457639fd691d669f2fe.

**Pull request merged**